plugin/completion.py

```diff
@@ -170,29 +156,28 @@ def initialize(self) -> None:
             # This is to make ST match with labels that have a weird prefix like a space character.
             settings.set("auto_complete_preserve_order", "none")
 
-    def on_text_changed(self, changes: Iterable[sublime.TextChange]) -> None:
-        if self._request_in_flight:
-            global _changes_since_sent
-            _changes_since_sent.extend(changes)
-
     def on_query_completions(self, prefix: str, locations: List[int]) -> Optional[sublime.CompletionList]:
         if not self.initialized:
             self.initialize()
         if not self.enabled:
             return None
-        client = client_from_session(session_for_view(self.view, 'completionProvider', locations[0]))
+        session = session_for_view(self.view, 'completionProvider', locations[0])
+        if not session:
+            return None
+        client = client_from_session(session)
         if not client:
             return None
```

Comment:
Is it possible that there is session but no client?

Reply:
Yes, when shutdown request is sent.

Reply:
Sorry, I mean when the exit notification has been sent out.